src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -1039,6 +1041,259 @@ class npc_magister_aledis : public CreatureScript
+        }
+
+        void SetData(uint32 /*type*/, uint32 data) override
+        {
+            switch (data)
+            {
+            case 1:
+                _events.ScheduleEvent(EVENT_ACTIVE_FALSE, Seconds(1));
+                break;
+            default:
+                break;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_LEONUS_TALK:
+                    {
+                        Talk(SAY_COVER);
+                        me->HandleEmoteCommand(EMOTE_ONESHOT_SHOUT);
+                        break;
+                    }
+                    case EVENT_INFERNAL_RAIN_ATTACK:
+                    {
+                        std::list<Creature*> InfernalRainList;
+                        Trinity::AllCreaturesOfEntryInRange checker_infernal(me, NPC_INFERNAL_RAIN, 200.0f);
+                        Trinity::CreatureListSearcher<Trinity::AllCreaturesOfEntryInRange> searcher_infernal(me, InfernalRainList, checker_infernal);
+                        Cell::VisitAllObjects(me, searcher_infernal, 200.0f);
+
+                        if (InfernalRainList.empty())
+                            return;
+
+                        for (std::list<Creature*>::iterator itr = InfernalRainList.begin(); itr != InfernalRainList.end(); ++itr)
+                            if (!(*itr)->isMoving() && (*itr)->GetPositionZ() > 118.0f)
+                                (*itr)->AI()->SetData(1, 1);
+
+                        break;
+                    }
+                    case EVENT_FEAR_CONTROLLER_CAST:
+                    {
+                        std::list<Creature*> FearControllerList;
+                        Trinity::AllCreaturesOfEntryInRange checker_fear(me, NPC_FEAR_CONTROLLER, 200.0f);
+                        Trinity::CreatureListSearcher<Trinity::AllCreaturesOfEntryInRange> searcher_fear(me, FearControllerList, checker_fear);
+                        Cell::VisitAllObjects(me, searcher_fear, 200.0f);
+
+                        if (FearControllerList.empty())
+                            return;
+
+                        for (std::list<Creature*>::iterator itr = FearControllerList.begin(); itr != FearControllerList.end(); ++itr)
+                            (*itr)->AI()->SetData(1, 1);
+
+                        break;
+                    }
+                    case EVENT_ACTIVE_FALSE:
+                    {
+                        _events.ScheduleEvent(EVENT_LEONUS_TALK, Hours(1));
+                        _events.ScheduleEvent(EVENT_INFERNAL_RAIN_ATTACK, Hours(1));
+                        _events.ScheduleEvent(EVENT_FEAR_CONTROLLER_CAST, Hours(1));
+                        break;
+                    }
+                }
+            }
+
+            if (!UpdateVictim())
```

Comment:
it should be on top of UpdateAI hook.
If it execute events while out of combat, control it using events phase :)
e.g. ```(!UpdateVictim() && !events.IsInPhase(XXXXXX))```